transport/transport.go

```diff
@@ -303,6 +303,24 @@ func StreamFromContext(ctx context.Context) (s *Stream, ok bool) {
 	return
 }
 
+// Peer connection info
+type Peer struct {
```

Comment:
okay, it seems it requires making changes to several code pieces to make this happen. Do you mind closing this PR and leaving this to me? I am trying to get it done by this Wed.